lib/src/async_environment.dart

```diff
@@ -670,6 +671,22 @@ class AsyncEnvironment {
     }
   }
 
+  /// Creates an implicit configuration from the variables declared in this
+  /// environment.
+  Configuration toImplicitConfiguration() {
+    var configuration = <String, ConfiguredValue>{};
+    for (var i = 0; i < _variables.length; i++) {
+      var values = _variables[i];
+      var nodes =
+          _variableNodes == null ? <String, AstNode>{} : _variableNodes[i];
+      for (var name in values.keys) {
+        configuration[name] =
+            ConfiguredValue(values[name], nodes[name]?.span, nodes[name]);
```

Comment:
Materializing `AstNode`s' spans can be expensive, which is why we pass them around as `AstNode`s in the first place, so I'd like to avoid `nodes[name]?.span` here. In practice this span is also never used, since an implicit configuration's variables are never invalid.

We could just pass `null` here, or we could encode this more statically by having a subclass with a constructor `ImplicitConfiguredValue(Value, AstNode)` and defining `Configuration.implicit(Map<String, ImplicitConfiguredValue>)`. Up to you.